### ISSUE TITLE
Segfault in Jhead at exif.c:323

### ISSUE THREAD

**Issue by Ngiong** [Matthias-Wandel/jhead#3]
Hello,
We are currently working on a new fuzz testing feature, and we found a crash in jhead.

**Step to Reproduce**
We used `CFLAGS="-g -O0" make` to compile the jhead, and run it with:
```
./jhead -v -ft -rgt -purejpg <attached_file> -se
```
Attachment: [poc_0002.txt](https://github.com/Matthias-Wandel/jhead/files/5420143/poc_0002.txt)

**Environment**
 - OS: Ubuntu 18.04.4 LTS
 - Compiler: gcc 7.5.0
 - jhead version: 3.04 (https://www.sentex.ca/~mwandel/jhead/jhead-3.04.tar.gz)

**Additional context**
Here's the stack trace: [stack_trace_0002.txt](https://github.com/Matthias-Wandel/jhead/files/5420358/stack_trace_0002.txt)

```
Program received signal SIGSEGV, Segmentation fault.
0x0000000000405ced in Get16u (Short=<optimized out>) at exif.c:323
323	exif.c: No such file or directory.
#0  0x0000000000405ced in Get16u (Short=<optimized out>) at exif.c:323
#1  PrintFormatNumber (ValuePtr=0x637000, Format=3, ByteCount=2) at exif.c:378
#2  0x0000000000408d61 in ProcessGpsInfo (DirStart=<optimized out>, OffsetBase=<optimized out>, ExifLength=<optimized out>) at gpsinfo.c:208
#3  0x000000000040717a in ProcessExifDir (DirStart=<optimized out>, OffsetBase=<optimized out>, ExifLength=<optimized out>, NestingLevel=<optimized out>) at exif.c:866
#4  0x0000000000407064 in ProcessExifDir (DirStart=<optimized out>, OffsetBase=<optimized out>, ExifLength=<optimized out>, NestingLevel=<optimized out>) at exif.c:852
#5  0x0000000000407064 in ProcessExifDir (DirStart=<optimized out>, OffsetBase=<optimized out>, ExifLength=<optimized out>, NestingLevel=<optimized out>) at exif.c:852
#6  0x0000000000407064 in ProcessExifDir (DirStart=<optimized out>, OffsetBase=<optimized out>, ExifLength=<optimized out>, NestingLevel=<optimized out>) at exif.c:852
#7  0x0000000000407064 in ProcessExifDir (DirStart=<optimized out>, OffsetBase=<optimized out>, ExifLength=<optimized out>, NestingLevel=<optimized out>) at exif.c:852
#8  0x0000000000406375 in process_EXIF (ExifSection=0x617500 "\002\020Exif", length=528) at exif.c:1041
#9  0x000000000040401d in ReadJpegSections (infile=0x6162c0, ReadMode=<optimized out>) at jpgfile.c:287
#10 0x00000000004046e8 in ReadJpegFile (FileName=0x7fffffffe23e "input", ReadMode=READ_ALL) at jpgfile.c:379
#11 0x000000000040240c in ProcessFile (FileName=<optimized out>) at jhead.c:905
#12 main (argc=<optimized out>, argv=<optimized out>) at jhead.c:1756
```

Thank you.

**Comment by Matthias-Wandel** [comment]
Fixed by change 56c108f1a0653b3a3c98960f66c2de5a9a67e5de on 
https://github.com/Matthias-Wandel/jhead

------ Original Message ------

**Comment by Matthias-Wandel** [comment]
Fixed by 56c108f1a0653b3a3c98960f66c2de5a9a67e5de
